include/moqt/publish_track_handler.h

```diff
@@ -0,0 +1,267 @@
+         *     period.
+         */
+        PublishTrackMetrics publish_track_metrics_;
+
+        // --------------------------------------------------------------------------
+        // Internals
+        // --------------------------------------------------------------------------
+      private:
+        /**
+         * @brief Publish Object function via the MoQ instance
+         *
+         * @details This is set by the MoQInstance.
+         *   Publish object function provides direct access to the MoQInstance that will publish
+         *   the object based on the track mode.
+         *
+         * @param priority              Priority to use for object; set on next track change
+         * @param ttl                   Expire time to live in milliseconds
+         * @param stream_header_needed  Indicates if group or track header is needed before this data object
+         * @param data                  Raw data/object that should be transmitted - MoQInstance serializes the data
+         */
+        using PublishObjFunction = std::function<PublishStatus(uint8_t priority,
+                                                               uint32_t ttl,
+                                                               bool stream_header_needed,
+                                                               uint64_t group_id,
+                                                               uint64_t object_id,
+                                                               BytesSpan data)>;
+        /**
+         * @brief Set the Data context ID
+         *
+         * @details The MOQ Handler sets the data context ID
+         */
+        void SetDataContextId(uint64_t data_ctx_id) { publish_data_ctx_id_ = data_ctx_id; };
+
+        /**
+         * @brief Get the Data context ID
+         */
+        uint64_t GetDataContextId() { return publish_data_ctx_id_; };
+
+        void SetPublishObjectFunction(PublishObjFunction&& publish_func)
+        {
+            publish_object_func_ = std::move(publish_func);
+        }
+
+        /**
+         * @brief Set the publish status
+         * @param status                Status of publishing (aka publish objects)
+         */
+        void SetStatus(Status status) { publish_status_ = status; }
+
+        // --------------------------------------------------------------------------
+        // Member variables
+        // --------------------------------------------------------------------------
+        Status publish_status_{ Status::kNotAnnounced };
+        TrackMode track_mode_;
+        uint8_t def_priority_;              // Set by caller and is used when priority is not specified
+        uint32_t def_ttl_;                  // Set by caller and is used when TTL is not specified
+
+        uint64_t publish_data_ctx_id_;      // publishing data context ID
+        PublishObjFunction publish_object_func_;
+
+        uint64_t object_payload_remaining_length_ { 0 };
```

Comment:
which object are we referring to here. Personally think there is a lot more internal structure being exposed in the public header/API. These will be better suited in a implementation class ?